packages/client/builtin/Toc.vue

```diff
@@ -0,0 +1,107 @@
+<!--
+Table Of content
+
+Usage:
+
+<Toc columns="2" maxDepth="3" mode="onlySiblings"/>
+-->
+<script lang="ts">
```

Comment:
Could we use the `<script setup>` syntax for overall consistency?

Reply:
Of course no problem, but I'm not completely familiar with this new syntax and I don't know how I can access the $slidev inside of it. Can you give me an hint ?

Reply:
`getCurrentInstance().proxy.$slidev` I think. But actually you can directly import them from the internal modules

Reply:
Thank you for your answer.
I have pushed the update.
Also what is your thought about the last 2 points ?
* add some styles to see active item in the list (classes are already added but I am not sure where I should put the styles or if we should let the user add his own styles)
* the tree for the TOC is a computed value inside the Toc component, but if you put multiple Toc component in your presentation, the tree get calculated for each components. Maybe we can calculate the tree globally and get it inside the component (that will filter it accordingly to the props), but I am not sure how we should do that.

For the last point can I add a computed value for the tree inside the `@slidev/client/logic/nav` ? Will it solves the problem ?

Reply:
1. I prefer to not have styles, but we could still add classes prefixed by `.slidev` just like other classes for users to override for certain state
2. I don't mind that as I don't think the computed will be very expensive. We could leave it as an future optimization if it really leads to performance issues